### PR TITLE
SSE inside the Resonator effect

### PR DESCRIPTION
**Pull request by baconpaul**
This isn't finished and is not on in config.xml, but this is
a complete implementation of how to use an SSE filter in the Resonator
Effect which @mkruselj can use to finish this effect.

### CODE REVIEW

File: src/common/dsp/effect/ResonatorEffect.cpp [hunk 1]
```diff
@@ -39,21 +46,152 @@ void ResonatorEffect::setvars(bool init)
...
+    {
+        for (int i = 0; i < n_filter_registers; i++)
+        {
+            for (int e = 0; e < 3; ++e)
+            {
+                Reg[e][c][i] = get1f(qfus[c].R[i], e);
+                coeff[e][c].C[i] = get1f(qfus[c].C[i], e);
+            }
+        }
+    }
```
Comment (mvf) [inline src/common/dsp/effect/ResonatorEffect.cpp:180]: GCC doesn't seem to like this part. Just pasting this here, didn't look into it.
```
../src/common/dsp/effect/ResonatorEffect.cpp: In member function 'virtual void ResonatorEffect::process(float*, float*)':
../src/common/dsp/effect/ResonatorEffect.cpp:243:41: warning: iteration 8 invokes undefined behavior [-Waggressive-loop-optimizations]
  243 |                 coeff[e][c].C[i] = get1f(qfus[c].C[i], e);
      |                                    ~~~~~^~~~~~~~~~~~~~~~~
../src/common/dsp/effect/ResonatorEffect.cpp:238:27: note: within this loop
  238 |         for (int i = 0; i < n_filter_registers; i++)
      |                         ~~^~~~~~~~~~~~~~~~~~~~
```

Comment (baconpaul) [reply src/common/dsp/effect/ResonatorEffect.cpp:180]: Oh man that’s a real bug. It should be r not c or n coefficient not registers. Will clean up today. Thanks. 

Which gcc version shows that? Surpsiede pipelines didn’t find

Comment (mvf) [reply src/common/dsp/effect/ResonatorEffect.cpp:180]: GCC 10.2.1 20201203. The pipelines also show this, but we're not using `-Werror` on GCC.

**Comment by baconpaul** [comment]
Ahh! Well two things to fix then! I thought we were werrror on linux now?

will get to it today. Basically just need two loops with different bounds there

File: src/common/dsp/effect/ResonatorEffect.cpp (hunk 1 above)
Comment (baconpaul) [reply src/common/dsp/effect/ResonatorEffect.cpp:180]: Anyway #3751 fixes both. Thank you!!!

**Pull request merged**